File: jvm/core/main/scala/com/webank/eggroll/core/containers/meta/StopContainersResponse.scala
package com.webank.eggroll.core.containers.meta

import com.google.protobuf.ByteString
import com.webank.eggroll.core.meta.Containers

import scala.language.implicitConversions


case class StopContainersResponse(sessionId: String) {
}

object StopContainersResponse {
  implicit def deserialize(byteString: ByteString): StopContainersResponse = {
    val proto = Containers.StopContainersResponse.parseFrom(byteString)
    StopContainersResponse(proto.getSessionId)
  }

  implicit def serialize(src: StopContainersResponse): Array[Byte] = {
    val builder = Containers.StopContainersResponse.newBuilder()
      .setSessionId(src.sessionId)
    builder.build().toByteArray
  }
}

File: jvm/core/src/main/java/org/fedai/eggroll/core/containers/meta/StopContainersResponse.java
package org.fedai.eggroll.core.containers.meta;

import com.google.protobuf.ByteString;
import com.google.protobuf.InvalidProtocolBufferException;
import com.webank.eggroll.core.meta.Containers;
import lombok.Data;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

@Data
public class StopContainersResponse {

    Logger log = LoggerFactory.getLogger(StopContainersResponse.class);

    private String sessionId;

    public StopContainersResponse(String sessionId) {
        this.sessionId = sessionId;
    }

    public StopContainersResponse deserialize(ByteString byteString) {
        Containers.StopContainersResponse proto = null;
        try {
            proto = Containers.StopContainersResponse.parseFrom(byteString);
        } catch (InvalidProtocolBufferException e) {
            log.error("StopContainersResponse.deserialize() error :", e);
        }
        return new StopContainersResponse(proto.getSessionId());
    }

    public byte[] serialize() {
        Containers.StopContainersResponse.Builder builder = Containers.StopContainersResponse.newBuilder()
                .setSessionId(this.sessionId);
        return builder.build().toByteArray();
    }
}